src/components/text-editor/prosemirror-adapter/prosemirror-adapter.tsx

```diff
@@ -263,12 +267,34 @@ export class ProsemirrorAdapter {
             plugins: [
                 ...exampleSetup({ schema: this.schema, menuBar: false }),
                 keymap(this.menuCommandFactory.buildKeymap()),
-                this.createMenuStateTrackingPlugin(this.actionBarItems),
                 createLinkPlugin(this.handleNewLinkSelection),
+                createMenuStateTrackingPlugin(
+                    editorMenuTypesArray,
+                    this.menuCommandFactory,
+                    this.updateActiveActionBarItems,
```

Comment:
NIT. The `updateCallback` that we use now would be hard to test as it explicitly relies on `this`. We could make the callback inside it pure, returning the action bar items and use something like 

```
createMenuStateTrackingPlugin(
    editorMenuTypesArray,
    this.menuCommandFactory,
    (activeTypes: Record<EditorMenuTypes, boolean>) => {
        this.actionBarItems = this.updateActiveActionBarItems(activeTypes, this.actionBarItems);
    },
)
```
And even put the callback outside the class in a separate file

The same can be said for `this.handleNewLinkSelection`

But my main reason for pointing this out is that the adapter file is growing quite large

Reply:
On that note, I would probably like to play around with the code for the link and see how much we can move into the plugin

Reply:
We will anyways be dependent on this on your anonymous arrow function, so I'm not so sure what we gain?